src/models/user.js

```diff
@@ -4,8 +4,8 @@ module.exports = User;
 
 function User(attr) {
 	// TODO: Remove this
-	attr.name = attr.nick;
-	attr.mode = attr.modes[0] || "";
+	attr.name = attr.name || attr.nick;
+	attr.mode = attr.mode || (attr.modes && attr.modes[0]) || "";
```

Comment:
I think this should have been fixed in the test, as irc-fw will only be providing us with modes array.